### DISCUSSION TITLE
2020 darcy_thermo_mech example: Windows line endings issue

### DISCUSSION THREAD

**Discussion by ColinMG2** [idaholab/moose#28783] [Q&A Getting Started]
Hello. I am trying to create a mesh for the darcy_thermo_mech example in the moose repository and when I run  ../darcy_therm
o_mech-opt -i step1_colin.i --mesh-only after defining my parameters in VSCode, I get this error: *** ERROR ***
/home/colinmoose/projects/moose/tutorials/darcy_thermo_mech/step01_diffusion/problems/step1_colin.i contains Windows(DOS) line endings which are not supported.

Stack frames: 10
0: libMesh::print_trace(std::ostream&)
1: moose::internal::mooseErrorRaw(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)
2: void mooseError<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&&)
3: MooseUtils::checkFileReadable(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool, bool, bool)
4: Parser::parse()
5: Moose::createMooseApp(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int, char**)
6: ../darcy_thermo_mech-opt(+0x30cf) [0x562f0b1c00cf]
7: /lib/x86_64-linux-gnu/libc.so.6(+0x29d90) [0x7f24dfd75d90]
8: __libc_start_main
9: ../darcy_thermo_mech-opt(+0x337a) [0x562f0b1c037a]
Abort(1) on node 0 (rank 0 in comm 0): application called MPI_Abort(MPI_COMM_WORLD, 1) - process 0

How can I fix this issue?

**Comment by GiudGiud** [comment]
if you are using VSCode you can press control shift P and select "change end of line sequence"
If you are on linux you can use `dos2unix` in the command line to clear the file of DOS line endings

**Reply by ColinMG2** [reply]
Okay. I tried changing the line endings from CRLF to LF and it still failed to work. It could be because the MOOSE Language server is failing to connect to the server. Here is the error message I received in VSCode: [Error - 4:28:32 PM] MOOSE Language Server client: couldn't create connection to server.
Launching server using command \\wsl.localhost\Ubuntu\home\colinmoose\projects\moose\tutorials\darcy_thermo_mech\step01_diffusion\darcy_thermo_mech-opt failed. Error: spawn \\wsl.localhost\Ubuntu\home\colinmoose\projects\moose\tutorials\darcy_thermo_mech\step01_diffusion\darcy_thermo_mech-opt ENOENT

**Reply by GiudGiud** [reply]
the languge server failures are not related I think

did the error change after the fix?
are you using WSL? if so, is your text editor installed on the windows side?

**Reply by ColinMG2** [reply]
I am using wsl in the command terminal on a Windows 11 OS. The error did not change even after changing the line endings in VSCode. I could try doing it in the ubuntu terminal using dos2unix but I would have to do sudo apt install dos2unix

**Reply by ColinMG2** [reply]
My text editor is for the Windows version of VSCode.

**Reply by GiudGiud** [reply]
> My text editor is for the Windows version of VSCode.

It probably saved again using CRLF right after you switched it. You ll have to change the settings to change the default line endings

> I could try doing it in the ubuntu terminal using dos2unix but I would have to do sudo apt install dos2unix

this would work. installing dos2unix is safe

**Reply by ColinMG2** [reply]
Okay I did dos2unix and that worked perfectly. This is my output now:  Mesh Information:
```
  elem_dimensions()={2}
  spatial_dimension()=2
  n_nodes()=1111
    n_local_nodes()=1111
  n_elem()=1000
    n_local_elem()=1000
    n_active_elem()=1000
  n_subdomains()=1
  n_elemsets()=0
  n_partitions()=1
  n_processors()=1
  n_threads()=1
  processor_id()=0
  is_prepared()=true
  is_replicated()=true

 Mesh Bounding Box:
  Minimum: (x,y,z)=(       0,        0,        0)
  Maximum: (x,y,z)=(      10,       20,        0)
  Delta:   (x,y,z)=(      10,       20,        0)

 Mesh Element Type(s):
  QUAD4

 Mesh Nodesets:
  Nodeset 0 (bottom), 101 nodes
   Bounding box minimum: (x,y,z)=(       0,        0,        0)
   Bounding box maximum: (x,y,z)=(      10,        0,        0)
   Bounding box delta: (x,y,z)=(      10,        0,        0)
  Nodeset 1 (right), 11 nodes
   Bounding box minimum: (x,y,z)=(      10,        0,        0)
   Bounding box maximum: (x,y,z)=(      10,       20,        0)
   Bounding box delta: (x,y,z)=(       0,       20,        0)
  Nodeset 2 (top), 101 nodes
   Bounding box minimum: (x,y,z)=(       0,       20,        0)
   Bounding box maximum: (x,y,z)=(      10,       20,        0)
   Bounding box delta: (x,y,z)=(      10,        0,        0)
  Nodeset 3 (left), 11 nodes
   Bounding box minimum: (x,y,z)=(       0,        0,        0)
   Bounding box maximum: (x,y,z)=(       0,       20,        0)
   Bounding box delta: (x,y,z)=(       0,       20,        0)

 Mesh Sidesets:
  Sideset 0 (bottom), 100 sides (EDGE2), 100 elems (QUAD4), 101 nodes
   Side volume: 10
   Bounding box minimum: (x,y,z)=(       0,        0,        0)
   Bounding box maximum: (x,y,z)=(      10,        0,        0)
   Bounding box delta: (x,y,z)=(      10,        0,        0)
  Sideset 1 (right), 10 sides (EDGE2), 10 elems (QUAD4), 11 nodes
   Side volume: 20
   Bounding box minimum: (x,y,z)=(      10,        0,        0)
   Bounding box maximum: (x,y,z)=(      10,       20,        0)
   Bounding box delta: (x,y,z)=(       0,       20,        0)
  Sideset 2 (top), 100 sides (EDGE2), 100 elems (QUAD4), 101 nodes
   Side volume: 10
   Bounding box minimum: (x,y,z)=(       0,       20,        0)
   Bounding box maximum: (x,y,z)=(      10,       20,        0)
   Bounding box delta: (x,y,z)=(      10,        0,        0)
  Sideset 3 (left), 10 sides (EDGE2), 10 elems (QUAD4), 11 nodes
   Side volume: 20
   Bounding box minimum: (x,y,z)=(       0,        0,        0)
   Bounding box maximum: (x,y,z)=(       0,       20,        0)
   Bounding box delta: (x,y,z)=(       0,       20,        0)

 Mesh Edgesets:
  None

 Mesh Subdomains:
  Subdomain 0: 1000 elems (QUAD4, 1000 active), 1111 active nodes
   Volume: 200
   Bounding box minimum: (x,y,z)=(       0,        0,        0)
   Bounding box maximum: (x,y,z)=(      10,       20,        0)
   Bounding box delta: (x,y,z)=(      10,       20,        0)
  Global mesh volume = 200
```
How can I fix the language server issues so I can have autofill in VSCode?

**Reply by GiudGiud** [reply]
you cannot fix it easily from Windows I suspect.
The easiest would be to install VSCode on the linux side, then once it finds a working executable for your application, it can generate the acceptable syntax list.

If you want to try to fix the windows VSCode plugin, then you need to generate the syntax dump in linux using `<your executable> --json`, then make VSCode on windows find that file. I m not sure how to do that last part

**Reply by ColinMG2** [reply]
I was able to fix the windows VSCode plugin by properly running in WSL: Ubuntu on VSCode. I now have the autofill capabilities from the MOOSE Language extension.
